### DISCUSSION TITLE
Esphome?

### DISCUSSION THREAD

**Discussion by dckiller51** [syssi/xiaomi_fan#150] [Ideas]
Hello, do you think that it is possible to create a firmware under Esphome. I disassembled the dmaker.fan.p18
![IMG_20210626_120326](https://user-images.githubusercontent.com/53062806/123511357-17999b00-d681-11eb-8204-2f2d76cd6423.jpg)
![IMG_20210626_120357](https://user-images.githubusercontent.com/53062806/123511359-18cac800-d681-11eb-8222-15add8168720.jpg)
![IMG_20210626_120347](https://user-images.githubusercontent.com/53062806/123511360-19635e80-d681-11eb-88c0-a231aed40014.jpg)

**Comment by syssi** [comment]
Cool! Let's do it! Do you need support?

**Reply by helgek** [reply]
@syssi I'd be more than happy do this with your support since you have already provided me with this amazing opportunity to now have 5 Yeelight and Xiaomi lamps running with ESPHome at my home! In addition I've recently added a Xiaomi Air Purifier 4 Lite to my smart home hardware selection which I also flashed with ESPHome ( https://github.com/dhewg/esphome-miot - probably it would then make much sense to follow-up with an issue for this fan there ).

I will receive the dmaker.fan.p18 / Mi Smart Standing Fan 2 next week.

I also posted about it here:
https://community.home-assistant.io/t/esphome-components-for-miot-devices/686646/5?u=helge_k

**Comment by helgek** [comment]
I shared my first steps attempting to build an ESPHome config for both the dmaker.fan.p18 (Mi Smart Standing Fan 2) and the dmaker.fan.p33 (Mi Smart Standing Fan 2 Pro) here: https://github.com/dhewg/esphome-miot/issues/19

**Reply by helgek** [reply]
@dckiller51 I just managed to flash a working ESPHome config with one test control (power on/off) and this works perfectly. I hope can soon complete the config and then we should soon have the first two fans that run on ESPHome.

https://github.com/dhewg/esphome-miot/issues/19
